react/react1/week3/todo-list/src/components/TodoList.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect, useCallback } from "react";
+import TodoItem from "./TodoItem";
+
+const TodoList = () => {
+  const API_URL =
+    "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+  const [todos, setTodos] = useState([]);
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+  // Fetch our todos
+  const fetchTodos = useCallback(() => {
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => {
+        setTodos(data);
+      });
+  }, []);
+
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
```

Comment:
Instead of having `useEffect `hook, giving `fetchTodos` callback function a dependency (todos) would work as you required here.
Is there any other thing that this `useEffect `is doing here?